tests/test_links.py

```diff
@@ -0,0 +1,190 @@
+    f.create_group('grp')
+    f['/grp/external'] = ExternalLink('mongoose.exdir', '/mongoose')
+    assert '/grp/external' in f
+    assert '/grp/external/something' not in f
+
+
+def test_get_link(setup_teardown_file):
+    """ Get link values """
+    f = setup_teardown_file[3]
+    g = File(setup_teardown_file[0] / 'somewhere.exdir')
+    f.create_group('mongoose')
+    g.create_group('mongoose')
+    sl = SoftLink('/mongoose')
+    el = ExternalLink('somewhere.exdir', 'mongoose')
+
+    f['soft'] = sl
+    f['external'] = el
+
+    out_sl = f.get('soft', get_link=True)
+    out_el = f.get('external', get_link=True)
+
+    assert isinstance(out_sl, SoftLink)
+    assert out_sl == sl
+    assert isinstance(out_el, ExternalLink)
+    assert out_el == el
+
+
+# Feature: Create and manage soft links with the high-level interface
+def test_soft_path(setup_teardown_file):
+    """ SoftLink directory attribute """
+    sl = SoftLink('/foo')
+    assert sl.path == '/foo'
+
+
+def test_soft_repr(setup_teardown_file):
+    """ SoftLink path repr """
+    sl = SoftLink('/foo')
+    assert isinstance(repr(sl), str)
+
+
+def test_linked_group_equal(setup_teardown_file):
+    """ Create new soft link by assignment """
+    f = setup_teardown_file[3]
+    g = f.create_group('new')
+    sl = SoftLink('/new')
+    f['alias'] = sl
+    g2 = f['alias']
+    assert g == g2
+
+
+def test_exc(setup_teardown_file):
+    """ Opening dangling soft link results in KeyError """
+    f = setup_teardown_file[3]
+    f['alias'] = SoftLink('new')
+    with pytest.raises(KeyError):
+        f['alias']
+
+
+# Feature: Create and manage external links
+def test_external_path(setup_teardown_file):
+    """ External link paths attributes """
+    external_path = setup_teardown_file[0] / 'foo.exdir'
+    g = File(external_path, 'w')
+    egrp = g.create_group('foo')
+    el = ExternalLink(external_path, '/foo')
+    assert el.filename == external_path
+    assert el.path == '/foo'
+
+
+def test_external_repr(setup_teardown_file):
+    """ External link repr """
+    external_path = setup_teardown_file[0] / 'foo.exdir'
+    g = File(external_path, 'w')
+    el = ExternalLink(external_path, '/foo')
+    assert isinstance(repr(el), str)
+
+
+def test_create(setup_teardown_file):
+    """ Creating external links """
+    external_path = setup_teardown_file[0] / 'foo.exdir'
+    f = setup_teardown_file[3]
+    g = File(external_path, 'w')
+    egrp = g.require_group('external')
+    f['ext'] = ExternalLink(external_path, '/external')
+    grp = f['ext']
+    ef = grp.file
+    assert ef != f
+    assert grp.name == '/external'
+
+
+def test_broken_external_link(setup_teardown_file):
+    """ KeyError raised when attempting to open broken link """
+    external_path = setup_teardown_file[0] / 'foo.exdir'
+    f = setup_teardown_file[3]
+    g = File(external_path, 'w')
+    f['ext'] = ExternalLink(external_path, '/missing')
+    with pytest.raises(KeyError):
```

Comment:
To add to the comment above: This should not be a KeyError. It should actually return the object that the link is pointing to.